src/EnergyPlus/DaylightingDevices.cc

```diff
@@ -1620,6 +1614,164 @@ namespace DaylightingDevices {
+        }
+        Real64 zWinMax = -9999.9;
+        Real64 zWinMin = 9999.9;
+        for (int vertex = 1; vertex <= 4; ++vertex) {
+            if (state.dataSurface->Surface(WinSurf).Vertex(vertex).z > zWinMax) zWinMax = state.dataSurface->Surface(WinSurf).Vertex(vertex).z;
+            if (state.dataSurface->Surface(WinSurf).Vertex(vertex).z < zWinMin) zWinMin = state.dataSurface->Surface(WinSurf).Vertex(vertex).z;
+        }
+
+        Real64 leftoverViewFactor;
+        // Now correct the view factors based on the location of the shelf with respect to the window
+        ShowWarningError(
+            state,
+            format("DaylightingDevice:Shelf = {}:  Window view factor to shelf [{:.2R}] results in a sum of view factors greater than 1.",
+                   state.dataDaylightingDevicesData->Shelf(ShelfNum).Name,
+                   state.dataDaylightingDevicesData->Shelf(ShelfNum).ViewFactor));
+        if (zWinMin >= zShelfMax) { // Shelf is fully below window, reduce view to ground first based on view to shelf
+            ShowContinueError(
+                state,
+                "Since the light shelf is below the window to which it is associated, the view factor of the window to the ground was reduced");
+            ShowContinueError(
+                state, "and possibly also the view factor to the sky. Check you input and/or consider turning off autosizing of the view factors.");
+            leftoverViewFactor = 1.0 - viewFactorToShelf - viewFactorToSky;
+            if (leftoverViewFactor >= 0.0) {
+                viewFactorToGround = leftoverViewFactor; // Other view factors okay
+            } else {
+                viewFactorToGround = 0.0;
+                viewFactorToSky = 1.0 - viewFactorToShelf;
+                if (viewFactorToSky < 0.0) {
+                    viewFactorToSky = 0.0;
+                    viewFactorToShelf = 1.0;
+                }
+            }
+
+        } else if (zShelfMin >= zWinMax) { // Shelf is fully above window, reduce view to sky first based on view to shelf
+            ShowContinueError(
+                state, "Since the light shelf is above the window to which it is associated, the view factor of the window to the sky was reduced");
+            ShowContinueError(
+                state,
+                "and possibly also the view factor to the ground. Check you input and/or consider turning off autosizing of the view factors.");
+            leftoverViewFactor = 1.0 - viewFactorToShelf - viewFactorToGround;
+            if (leftoverViewFactor >= 0.0) {
+                viewFactorToSky = leftoverViewFactor;
+            } else {
+                viewFactorToSky = 0.0;
+                viewFactorToGround = 1.0 - viewFactorToShelf;
+                if (viewFactorToGround < 0.0) {
+                    viewFactorToGround = 0.0;
+                    viewFactorToShelf = 1.0;
+                }
+            }
```

Comment:
So for after all done for this case, maybe a little bit extra (optional) warning the user should still check the view from the window to the building itself which might complicate the case and use manual input when applicable(?).

Reply:
I will add another warning at the end of this routine that specifically mentions this and covers all of the cases where something had to be reduced.